testdata/go/v3/memcached-operator/tools/metricsdocs/metricsdocs.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/example/memcached-operator/monitoring/metrics"
+	"github.com/example/memcached-operator/monitoring/metrics/util"
+)
+
+// constant parts of the file
+const (
+	title      = "# Memcached Operator Metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by the memcached-operator.\n" +
+		"All metrics documented here are auto-generated by the utility tool `tools/metricsdocs` and reflects exactly what is being exposed.\n\n"
+
+	KVSpecificMetrics = "## Memcached Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "In order to add a new metric follow this steps:\n" +
+		"- Make sure the metric type defined in ./monitoring/metrics/util/util.go,\n" +
+		"  as for now the types available are counter, and gauge.\n" +
+		"- Add a new metric under ./monitoring/metrics\n" +
+		"  e.g. ./monitoring/metrics/reconcile_metrics.go\n" +
+		"- add the metric under ./monitoring/metrics/metrics.go\n\n" +
+		"After developing new metrics or changing old ones, please run `make generate-metricsdocs` to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change `tools/metricsdocs` with your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+// MemcachedMetricList contains the name, description, and type for each metric.
+func main() {
+	MemcachedMetricList := metricDescriptionListToMetricList(metrics.ListMetrics())
+	sort.Sort(MemcachedMetricList)
+	writeToFile(MemcachedMetricList)
+}
+
+// writeToFile receives a list of metrics and writes them to a file.
+func writeToFile(metricsList metricList) {
+	fmt.Print(opening)
+	metricsList.writeOut()
+	fmt.Print(footer)
+}
+
+// metric is an exported struct that defines the metric
+// name, description, and type as a new type named metric.
+type metric struct {
```

Comment:
Not sure type metric is needed since util.Metric includes the same fields.
@machadovilaca what do you think?